packages/api/src/car.js

```diff
@@ -263,41 +261,77 @@ export async function sizeOf (response) {
 }
 
 /**
- * Returns the sum of all block sizes in the received CAR. Throws if any block
+ * @param {CarReader} reader
+ * @param {CarStat} stat
+ */
+async function validateCar (reader, stat) {
```

Comment:
What is the case that leads to the creation of such CAR files? I am not sure if this is the best place for such validation?

Reply:
go-ipfs can create them where it's using a url-store and the url can no longer be reached. it's an edge case that we hit pretty hard, but is otherwise pretty uncommon.